Wdb 82/setup docker containers for develop

kibartas opened the pull request:
AMA about the changes. 

Check that the first way to run backend works, at least, the other two are auxiliary.

The Dockerfile is to create an image from the API, the docker-compose.yml is to run both API and DB in two separate containers at the same time.

The Startup.cs change is QOL. No need to `update-database` anymore. (Also helps to keep it simple with the API Docker image)

AArlauskas committed:
[WDB-82]
Rollback appsettings.json

Commit:
add Dockerfile for ASP.NET Core project image building and docker-compose.yml file to setup DB + ASP.NET Core for development

Commit:
add a method to Startup.cs that removes the need to 'dotnet-ef database update'

Commit:
change database connection string to work with Linux MSSQL (no Windows auth)

Commit:
[WDB-82]
merge ABC-123

Commit:
[WDB-82]
bugfix: prepare-commit-msg needs yarn prepended to it's git hook

Commit:
[WDB-82]
improve README.md

Commit:
Merge remote-tracking branch 'origin/master' into WDB-82/Setup-Docker-containers-for-develop

Commit:
Merge remote-tracking branch 'origin/ABC-1234/husky-prepare-commit-msg-hook' into WDB-82/Setup-Docker-containers-for-develop

AArlauskas commented on backend/backend/appsettings.json:
```diff
@@ -1,6 +1,6 @@
 {
   "ConnectionStrings": {
-    "BackendContext": "Server=.\\SQLEXPRESS;Database=WeDontByte;Trusted_Connection=True;"
+    "BackendContext": "Server=.\\SQLEXPRESS,1433;Database=WeDontByte;User=sa;Password=Password@123;"
```
This throws exceptions for me. I created a separate user and it works if you do not provide the port. For SQLEXPRESS 1433 is the default port. Is is really necessary to give it as a param in connection string?

kibartas replied:
https://social.msdn.microsoft.com/Forums/sqlserver/en-US/e1082cad-9248-44ed-bfcc-c08d7cc08831/login-failed-for-user-sa-microsoft-sql-server-error-18456-severity-14-state-1-when-i-use


Check this link. The port is mandatory for Docker, it seems

kibartas replied:
Reverted the change completely, should work without hiccups in Windows

AArlauskas commented on README.md:
```diff
@@ -2,15 +2,21 @@
\ No newline at end of file
+Second way:
+1. Have docker-compose installed
+2. Run `sudo docker-compose up` to launch both DB and backend at the same time (ignore the errors)
```
Ignore the errors sound nice 😆

kibartas replied:
Xdd they're really not a big deal

AArlauskas commented on README.md:
```diff
@@ -2,15 +2,21 @@
+Second way:
+1. Have docker-compose installed
+2. Run `sudo docker-compose up` to launch both DB and backend at the same time (ignore the errors)
+3. The backend will be running at http://localhost:5000
```
Can it run on the same port as the IIS (44344)? Then we wouldn't need to change keep changing endpoints in frontend

kibartas replied:
Yes, no problem

AArlauskas reviewed (commented):
Looks good, but need to get the answers and fix the connection to the DB

Commit:
[WDB-82]
change backend port to 44344 and remove SQLEXPRESS from the connection string

Commit:
[WDB-82]
update README.md and change Dockerfile's sed command to replace the connection string completely, regardless of what it is

kibartas committed:
Merge branch 'master' into WDB-82/Setup-Docker-containers-for-develop

Commit:
[WDB-82]
merge master

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
[WDB-82]
merge master

Commit:
Merge branch 'WDB-82/Setup-Docker-containers-for-develop' of github.com:kibartas/PSK-Front-End into WDB-82/Setup-Docker-containers-for-develop

tmaciulis22 commented on README.md:
```diff
@@ -2,15 +2,22 @@
 1. You must have SQLEXPRESS installed.
 2. Make sure that your SQLEXPRESS connection string matches the one in appsettings.json.
-3. Open up CMD or use Package manager console to run command "update-database".
 4. Start the backend project.
```
```suggestion
3. Start the backend project.
```

tmaciulis22 reviewed (approved):
Minor change. Did not try to run the backend yet, but it seems that everything will work fine, when I'll try it

kibartas committed:
Update README.md

Co-authored-by: Tautvydas Mačiulis <maciulis.tautvydas22@gmail.com>
